runtime/near-vm-runner/src/near_vm_runner.rs

```diff
@@ -676,61 +733,36 @@ impl<'a> finite_wasm::wasmparser::VisitOperator<'a> for GasCostCfg {
 impl crate::runner::VM for NearVM {
     fn run(
         &self,
-        code: &ContractCode,
+        code_hash: CryptoHash,
+        code: Option<&ContractCode>,
         method_name: &str,
         ext: &mut dyn External,
-        context: VMContext,
+        context: &VMContext,
         fees_config: &RuntimeFeesConfig,
         promise_results: &[PromiseResult],
         cache: Option<&dyn ContractRuntimeCache>,
     ) -> Result<VMOutcome, VMRunnerError> {
-        let mut memory = NearVmMemory::new(
```

Comment:
I'm trying to understand why this needed to be moved in this PR? Is this largely just a "makes code not need the `Outcome` hack anymore" or is there something deeper here?

Reply:
It was both the `enum Outcome` removal, and the fact that we need access to the `VMLogic` in order to charge for contract loading. But contract loading having entirely been moved to `with_compiled_and_loaded`, at some point during the refactor I had to get access to `VMLogic` from there, and then the move just happened.